video/out/vo.c

```diff
@@ -980,13 +980,25 @@ static bool render_frame(struct vo *vo)
 
         vo->driver->flip_page(vo);
 
+        int64_t forced_wait = 0;
+        if (vo->opts->display_fps_limiter && VS_IS_DISP(vo->opts->video_sync)) {
+            forced_wait = now + in->nominal_vsync_interval;
+            wait_until(vo, forced_wait);
```

Comment:
I think you wanted to sleep before flip page. Also this will not be synchronized with vsync, so we will miss vsync points, causing stuttering, not sure how it solves things.